app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java

```diff
@@ -167,9 +168,9 @@ private void loadData() {
                         final Connector existing = fetch(Connector.class, id);
                         if (existing != null) {
                             // the only mutable part of the Connector
-                            final Map<String, String> configuredProperties = existing.getConfiguredProperties();
-
-                            connector = connector.builder().configuredProperties(configuredProperties).build();
+                            final Map<String, String> existingConfiguredProperties = existing.getConfiguredProperties();
+                            final Map<String, String> mergedConfiguredProperties = merge(existingConfiguredProperties, connector.getConfiguredProperties());
+                            connector = connector.builder().configuredProperties(mergedConfiguredProperties).build();
```

Comment:
I've upgraded Immutables for #8269 and there's a `putAllConfiguredProperties` method on `Builder` classes, not sure if it's there in the current version. Might want to wait for that as it makes the implementation simpler:

```java
connector = connector.builder().putAllConfiguredProperties(existing.getConfiguredProperties())
```

Reply:
Unfortunately it fails the `shouldOverrideExistingConfiguredProperties` test. Basically it does not override any new configured properties introduced in the new connector version.